modules/Microsoft.ContainerRegistry/registries/deploy.bicep

```diff
@@ -51,6 +51,23 @@ param retentionPolicyStatus string = 'enabled'
 @description('Optional. The number of days to retain an untagged manifest after which it gets purged.')
 param retentionPolicyDays int = 15
 
+@allowed([
+  'disabled'
+  'enabled'
+])
+@description('Optional. The value that indicates whether the policy for using ARM audience token for a container registr is enabled or not.')
+param azureADAuthenticationAsArmPolicyStatus string = 'enabled'
+
+@allowed([
+  'disabled'
+  'enabled'
+])
+@description('Optional. Soft Delete policy status.')
+param softDeletePolicyStatus string = 'enabled'
```

Comment:
The default implementation does not have this enabled by default.

Reply:
@ahmadabdalla can you elaborate which default implementation you are referring to? I need to check if there is a security baseline recommendation too.

Reply:
A vanilla portal experience doesn't explicitly enable this feature. So if we are going to set this to default, we have to call it out. Maybe just a reference in the parameter description. Especially that we're also going to have to select a default time period, so what's the upper and lower limit to that as well.

Reply:
Just noticed that this feature has just came out in public preview 

https://azure.microsoft.com/en-au/updates/public-preview-soft-delete-in-azure-container-registry/

Maybe we shouldn't enable it by default given that customers might use it out of the box not knowing it isn't a supported capability?

Reply:
i set it to `disabled` now to align with the portal experience. I checked PSRule for Azure baseline, right now, it doesn't look like there's a rule created for this since it's still in preview. maybe @BernieWhite could create a rule for ACR Soft Delete feature and we can set the default value of this parameter to `enabled` in a future PR?

Reply:
@tyconsulting Nice thought. Yes it's a good idea to have a rule for this.